packages/mobile/src/account/Settings.test.tsx

```diff
@@ -60,4 +70,33 @@ describe('Account', () => {
     )
     expect(tree).toMatchSnapshot()
   })
+
+  it('navigates to PincodeSet screen if entered PIN is correct', async () => {
+    const tree = render(
+      <Provider store={createMockStore({})}>
+        <Settings {...getMockStackScreenProps(Screens.Settings)} />
+      </Provider>
+    )
+    fireEvent.press(tree.getByTestId('ChangePIN'))
+    mockedNavigate.mockImplementationOnce((_, params) => {
```

Comment:
This test (and the next) is not really doing what we want them to.

We're writing the mock implementation of the `navigate` function _after_ pressing the button. We should do that before. It will run automatically when the button is pressed. After pressing the button we should only write `expect`s to verify that what we expected to happen happened.

Since we're mocking the `ensurePincode` we just need to provide the logic we want `ensurePincode` to have. In this test, we want it to succeed by returning a promise that has `true` as its argument (the `true` is the `pinIsCorrect` variable).

So, it should be something like:

```
const tree = (...)
mockedEnsurePincode.mockImplementationOnce(() => Promise.resolve(true))
fireEvent.press(tree.getByTestId('ChangePIN'))
expect(navigate).toHaveBeenCalledWith(arguments)
```

For the next test, instead of returning a promise that resolves with `true` we should throw an error, which will cause the `ensurePincode` call to go the the `catch` callback.

Reply:
ooooooo thank you gonza! this makes a lot of sense, will rewrite